src/Microsoft.DotNet.Build.CloudTestTasks/CreateAzureContainer.cs

```diff
@@ -77,18 +82,49 @@ public override bool Execute()
 
         public async Task<bool> ExecuteAsync()
         {
+            if (!string.IsNullOrEmpty(ConnectionString))
```

Comment:
It's unfortunate to see this code duplicated so often.  Perhaps consider creating a ConnectionStringBasedTask class which is derived from Task and contains some of the shared logic.